x/protorev/keeper/rebalance.go

```diff
@@ -0,0 +1,160 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+// IterateRoutes checks the profitability of every single route that is passed in
+// and returns the optimal route if there is one
+func (k Keeper) IterateRoutes(ctx sdk.Context, routes []gammtypes.SwapAmountInRoutes) (sdk.Coin, sdk.Int, gammtypes.SwapAmountInRoutes) {
```

Comment:
Asking to learn: How does the system identifies routes? Do we have code for that logic yet?

Reply:
@stackman27 We have two methods of creating routes, a highest liquidity method and a hot routes method.

Our ReadMe draft explains the methodology in the state transitions section: https://skip-protocol.notion.site/x-protorev-Spec-3ba5cb368f314120bc514eb2d0a6a0dd

The relevant code pushed for this is in epoch_hook.go and routes.go.

Highest liquidity methods finds the highest liquidity atom pools and osmo pools and uses them to be the first and last hop of a cycle, where we place the pool the user swapped in the middle of the cycle.

The hot routes method is uploaded by us based on external analysis of routes searchers use / have arb opportunities often.

Reply:
Gotchu this makes sense